component/main.jsonnet

```diff
@@ -25,8 +25,26 @@ local memberlist_secret = kube.Secret(params.speaker.secretname) {
   },
 };
 
+local configmap = kube.ConfigMap(params.configmap_name) {
+  metadata+: {
+    namespace: params.namespace,
+  },
+  data: {
+    config: std.manifestYamlDoc(if params.config == null then {
+      "address-pools": [
+        {
+          "name": "default",
+          "protocol": "layer2",
+          "addresses": params.addresses,
```

Comment:
Two options here:
* Default `parameters.metallb.addresses` to the empty list `[]`
* Default `parameters.metallb.addresses` to `null` and throw an actual error using the `error` keyword when both `parameters.metallb.addresses and `parameters.metallb.config` are `null`.

The current code will result in Jsonnet stack traces if neither addresses nor config are set. The stack traces won't help the average user in resolving their configuration error.

I'm in favor of erroring out when neither addresses nor custom config are set, as silently deploying metallb with no addresses configured will also lead to confusion down the line.

"Fail early, fail loud" should be a consideration when writing components (not that the existing components are a shining example yet).

Reply:
Completely agree, that an proper error handling improves the overall usability. If no VIP is provided, this will lead to a broken MetallB. Just the question, if we will introduce bootstrap issues, when we hard fail. I think we will figure out this and for now let's try the "Fail early, fail loud" way.

Probably we should take this guideline as a guideline for error messages: https://uxplanet.org/how-to-write-good-error-messages-858e4551cd4